src/taipy/core/scenario/_scenario_converter.py

```diff
@@ -40,10 +44,18 @@ def _entity_to_model(cls, scenario: Scenario) -> _ScenarioModel:
 
     @classmethod
     def _model_to_entity(cls, model: _ScenarioModel) -> Scenario:
+        tasks = None
+        if model.tasks:
+            tasks = set(model.tasks)
+        else:
+            if model.pipelines:
+                tasks = Scenario._get_set_of_tasks_from_pipelines(model.pipelines)
```

Comment:
please move this part to the scenario model module. And isolate it in a dedicated method.
You have an example of migration functions in `data_node_model` module.